src/plugins/embeddable/public/embeddable_panel/index.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+
+import { EmbeddablePanelProps } from './types';
+import { ErrorEmbeddable, IEmbeddable } from '../lib';
+import { useEmbeddablePanel } from './use_embeddable_panel';
+import { EmbeddableLoadingIndicator } from './embeddable_loading_indicator';
+
+/**
+ * Loads and renders an embeddable.
+ */
+export const EmbeddablePanel = (props: EmbeddablePanelProps) => {
```

Comment:
Instead of both `EmbeddablePanel` and `EmbeddablePanelAsync`, how about just `EmbeddablePanel` and only `getEmbeddable: () => Promise<IEmbeddable | ErrorEmbeddable>;`. That would remove complexity and not really have any performance impact since `useEmbeddablePanel` still has to wait for other promises.

Reply:
Good call @nreese! I used a `union` type to combine both components into one, where the `embeddable` prop can either be passed as an unwrapped Embeddable, or as an async function that returns an Embeddable.